app/kuma-dp/pkg/dataplane/meshmetrics/component.go

```diff
@@ -68,6 +70,14 @@ func (cf *ConfigFetcher) Start(stop <-chan struct{}) error {
 		"socketPath", fmt.Sprintf("unix://%s", cf.socketPath),
 	)
 
+	ctx, ctxCancel := context.WithCancel(context.Background())
+	go func() {
+		select {
+		case <-stop:
+			ctxCancel()
+		}
```

Comment:
Nit, feel free to ignore but

```suggestion
		_ <-stop
		ctxCancel()
```

Reply:
changed